specifyweb/frontend/js_src/lib/utils/functools.ts

```diff
@@ -114,6 +114,12 @@ export const f = {
    * Useful when mapping over a list of functions
    */
   call: <RETURN>(callback: () => RETURN): RETURN => callback(),
+  /**
+   * Given an array of objects, return an array containing the values
+   * of a given key from each object in the array
+   */
+  pluck: <T extends RA<IR<any>>>(array: T, key: string): RA<any> =>
```

Comment:
1. Don't use `any`. This function can be written in a type-safe way:

```typescript
const pluck = <KEY extends PropertyKey, VALUE>(objects: RA<RR<KEY,VALUE>>, key: KEY): RA<VALUE> =>
  objects.map((obj) => obj[key]);
```

2. (optional) It's questionable whether pluck is even needed anymore. It was necessary in underscore because there was no native `.map` method, but now it's quite useless. Plus, it would be confusing for people not familiar with what `pluck` does, where as just having `array.map((obj) => obj[key])` is very clear

Reply:
> Don't use `any`. This function can be written in a type-safe way:
> 
> ```ts
> const pluck = <KEY extends PropertyKey, VALUE>(objects: RA<RR<KEY,VALUE>>, key: KEY): RA<VALUE> =>
>   objects.map((obj) => obj[key]);
> ```

Thanks for taking the time to write an example! I was wondering how I would go about making it type safe, and examples like these help me tighten the TypeScript screw in my brain!


> (optional) It's questionable whether pluck is even needed anymore. It was necessary in underscore because there was no native .map method, but now it's quite useless. Plus, it would be confusing for people not familiar with what pluck does, where as just having array.map((obj) => obj[key]) is very clear

I can very easily remove pluck then. Both from the reasons you listed and the fact it would be used in two locations